addon/services/engine-router-service.js

```diff
@@ -0,0 +1,118 @@
+import Service from '@ember/service';
+import { assert } from '@ember/debug';
+import { action, computed, get } from '@ember/object';
+import { reads } from '@ember/object/computed';
+import { getOwner } from '@ember/application';
+import Evented from '@ember/object/evented';
+import { namespaceEngineRouteName } from '../utils/namespace-engine-route-name';
+import { getRootOwner } from '../utils/owner';
+
+export default class EngineRouterService extends Service.extend(Evented) {
+  init() {
+    super.init();
+
+    this._externalRoutes = getOwner(this)._externalRoutes;
+    this._mountPoint = getOwner(this).mountPoint;
+    this.rootApplication = getRootOwner(this);
+
+    this.externalRouter.on('routeWillChange', this.onRouteWillChange);
+    this.externalRouter.on('routeDidChange', this.onRouteDidChange);
+  }
+
+  destroy() {
+    this.externalRouter.off('routeWillChange', this.onRouteWillChange);
+    this.externalRouter.off('routeDidChange', this.onRouteDidChange);
+
+    super.destroy();
+  }
+
+  @action
+  onRouteWillChange(...args) {
+    this.trigger('routeWillChange', ...args);
+  }
+
+  @action
+  onRouteDidChange(...args) {
+    this.trigger('routeDidChange', ...args);
+  }
+
+  @reads('externalRouter.rootURL') rootURL;
+
+  @reads('externalRouter.currentURL') currentURL;
+
+  @computed('externalRouter.currentRouteName')
+  get currentRouteName() {
+    if (get(this, 'externalRouter').currentRouteName === this._mountPoint) {
+      return 'application';
+    }
+    return get(this, 'externalRouter').currentRouteName.slice(this._mountPoint.length + 1);
```

Comment:
This should probably throw an assertion error, if the current route does not start with the mount point.

Reply:
```suggestion
    return this.externalRouter.currentRouteName.slice(this._mountPoint.length + 1);
```

Reply:
when `this._mountPoint` will be `null`? I can't see this scenario since the `mountPoint` is always required to engines - https://github.com/emberjs/ember.js/blob/59f1d40ada46475eff686ca38e93407e533273ee/packages/%40ember/-internals/glimmer/lib/component-managers/outlet.ts#L105